pipeline/authn/authenticator_cookie_session.go

```diff
@@ -49,6 +52,17 @@ func NewAuthenticatorCookieSession(c configuration.Provider) *AuthenticatorCooki
 	}
 }
 
+func (a *AuthenticatorCookieSessionConfiguration) ToAuthenticatorForwardConfig() *AuthenticatorForwardConfig {
+	return &AuthenticatorForwardConfig{
+		CheckSessionURL:    a.CheckSessionURL,
+		PreserveQuery:      a.PreserveQuery,
+		PreservePath:       a.PreservePath,
+		PreserveHost:       a.PreserveHost,
+		ForwardHTTPHeaders: a.ForwardHTTPHeadersMap,
```

Comment:
It's confusing that we have `ForwardHTTPHeaders` and `ForwardHTTPHeadersMap`, but depending on where this field is used they are used differently. To help others reading your code, it is good to be explicit about it. As mentioned in a comment below, I don't think we need the map at all.

Reply:
Without a map, it's easy to have duplicated keys for "Authorization" and "Cookie" headers for backward compatibility. Maps guarantee that all headers are unique. Using `map[string]struct{}` or `map[string]bool` can save space. On the other hand, the oathkeeper needs to make additional checks for backward compatibility which means more `for` loops at the initialization.

Reply:
I can unify variable names to make the code easier to understand.